2_segmenting_data/segmentation_utils.py

```diff
@@ -0,0 +1,251 @@
+    gfp_id = f"{field_details[0]}_{field_details[1]}_2_{field_details[3]}"
+    rfp_id = f"{field_details[0]}_{field_details[1]}_3_{field_details[3]}"
+
+    # Iterate through the directory containing the images and if an image has the ID of the given channel, it will read in that image. There are three images total
+    # that are read in for each site that needs to be overlayed
+    for image_path in images_dir.iterdir():
+        if dapi_id in image_path.name:
+            dapi = io.imread(image_path)
+        if gfp_id in image_path.name:
+            gfp = io.imread(image_path)
+        if rfp_id in image_path.name:
+            rfp = io.imread(image_path)
+
+    # RFP = red channel (cytoplasm), GFP = green channel (ER), DAPI = blue (nuclei)
+    # Multiplier only assists in user interpretation of the images to configure model parameters in CellPose GUI
+    overlay = np.dstack([rfp * 3, gfp, dapi])
+
+    return overlay
+
+
+def get_nuc_cyto_data(
+    nuc_locations: pd.DataFrame, cyto_outlines: pd.DataFrame
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """creates nuclei and cytoplasm dataframes with corresponding cell IDs from nuclei locations and cytoplasm outlines
+    Args:
+        nuc_locations (pd.DataFrame): dataframe with nuclei center coords for certain image
+        cyto_outlines (pd.DataFrame): dataframe with cytoplasm outline for same image as above
+    Returns:
+        Tuple[pd.DataFrame, pd.DataFrame]: nuc_data and cyto_data with corresponding cell IDs and center coords for each object
+    """
+    # Empty lists for the nuclei and cytoplasm data
+    nuc_data = []
+    cyto_data = []
+
+    # add Cell_ID column from index to the cyto_outlines dataframe where each each index value is the Cell_ID number
+    cyto_outlines = cyto_outlines.reset_index()
+    cyto_outlines = cyto_outlines.rename(columns={"index": "Cell_ID"})
+
+    # Iterate over all cytoplasm outlines
+    for cyto_index, cyto_row in cyto_outlines.iterrows():
+        cell_id = cyto_row["Cell_ID"]
+
+        # Create path (polygon) from outlines that is used to check if the nuclei are in this path
+        outline = cyto_row["Outline"]
+        cytoplasm_path = mplPath.Path(outline)
+
+        # Check if any of the nuclei are within cytoplasm path
+        for nuc_index, nuc_row in nuc_locations.iterrows():
+            nuc_center = (nuc_row["Location_Center_X"], nuc_row["Location_Center_Y"])
+            # If nuclei (one or more) is in cytoplasm path, this gives it same cell ID as cytoplasm that it is within
+            if cytoplasm_path.contains_point(nuc_center):
+                current_nuc_data = {
+                    "Cell_ID": cell_id,
+                    "Location_Center_X": nuc_row["Location_Center_X"],
+                    "Location_Center_Y": nuc_row["Location_Center_Y"],
+                }
+                nuc_data.append(current_nuc_data)
+
+                # Drop nucleus once it has been identfied in a cytoplasm so isn't rechecked
```

Comment:
b/c it is iterating over rows, doesn't this actually have no effect? ohhhh, wait, it does! good call. b/c you're iterating over `nuc_locations` while iterating over `cyto_outlines`.

I guess then my question becomes, what happens with overlapping cells?

Reply:
Already discussed. This should not be an issue because only the center (x,y) is being used so one pixel should end up in one cytoplasm.